File: src/test/java/com/fluxchess/pulse/RankTest.java
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

import org.junit.Test;

import java.lang.reflect.InvocationTargetException;

import static com.fluxchess.pulse.Rank.NORANK;
import static com.fluxchess.test.AssertUtil.assertUtilityClassWellDefined;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class RankTest {

  @Test
  public void testUtilityClass()
      throws InvocationTargetException, NoSuchMethodException, InstantiationException, IllegalAccessException {
    assertUtilityClassWellDefined(Rank.class);
  }

  @Test
  public void testValues() {
    for (int rank : Rank.values) {
      assertThat(Rank.values[rank], is(rank));
    }
  }

  @Test
  public void testIsValid() {
    for (int rank : Rank.values) {
      assertThat(Rank.isValid(rank), is(true));
    }

    assertThat(Rank.isValid(NORANK), is(false));
  }

}

File: src/test/cpp/ranktest.cpp
/*
 * Copyright (C) 2013-2016 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "rank.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(ranktest, testValues) {
	for (auto rank : Rank::values) {
		EXPECT_EQ(rank, Rank::values[rank]);
	}
}
